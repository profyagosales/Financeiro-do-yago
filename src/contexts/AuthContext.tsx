import type { User } from '@supabase/supabase-js';
import { createContext, useContext, useEffect, useState } from 'react';

import { supabase } from '@/lib/supabaseClient';

export type AuthContextValue = {
  user: User | null;
  loading: boolean;
  signOut: () => Promise<void>;
  // expostos para o Login.tsx
  loginWithPassword: (email: string, password: string) => Promise<void>;
  signUpWithPassword: (email: string, password: string) => Promise<void>;
};

const AuthContext = createContext<AuthContextValue | undefined>(undefined);

export function AuthProvider({ children }: { children: React.ReactNode }) {
  const [user, setUser] = useState<User | null>(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    let mounted = true;

    supabase.auth.getSession().then(({ data }) => {
      if (!mounted) return;
      setUser(data.session?.user ?? null);
      setLoading(false);
    });

    const {
      data: { subscription },
    } = supabase.auth.onAuthStateChange((_event, session) => {
      setUser(session?.user ?? null);
      setLoading(false);
    });

    return () => {
      mounted = false;
      subscription.unsubscribe();
    };
  }, []);

  const loginWithPassword = async (email: string, password: string) => {
    const { error } = await supabase.auth.signInWithPassword({ email, password });
    if (error) throw error;
  };

  const signUpWithPassword = async (email: string, password: string) => {
    const { error } = await supabase.auth.signUp({
      email,
      password,
      options: { emailRedirectTo: `${window.location.origin}/confirm` },
    });
    if (error) throw error;
  };

  const signOut = async () => {
    await supabase.auth.signOut();
  };

  const value: AuthContextValue = {
    user,
    loading,
    signOut,
    loginWithPassword,
    signUpWithPassword,
  };

  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
}

export function useAuth() {
  const ctx = useContext(AuthContext);
  if (!ctx) throw new Error('useAuth deve ser usado dentro de <AuthProvider>');
  return ctx;
}
