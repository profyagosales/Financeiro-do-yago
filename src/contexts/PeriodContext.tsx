/**
 * src/contexts/PeriodContext.tsx
 * Shim de compatibilidade: reexporta a implementação real do estado.
 * Assim, imports antigos de "@/contexts/PeriodContext" seguem funcionando
 * mesmo que a fonte oficial esteja em "@/state/periodFilter".
 */
export { PeriodProvider, usePeriod } from "@/state/periodFilter";
