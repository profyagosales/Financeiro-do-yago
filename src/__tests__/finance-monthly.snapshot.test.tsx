import { render } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { describe, expect, it } from 'vitest';

import { AuthProvider } from '@/contexts/AuthContext';
import { PeriodProvider } from '@/contexts/PeriodContext';
import FinanceMonthly from '@/pages/FinanceMonthly';

// Snapshot inicial da página mensal (estrutura básica + filtros + KPI placeholders)
// Evita regressões visuais grosseiras durante refactors.

describe('FinanceMonthly snapshot', () => {
  it('matches structure', () => {
    const { asFragment } = render(
      <AuthProvider>
        <PeriodProvider>
          <MemoryRouter initialEntries={['/financas/mensal?mes=2025-08']}>
            <Routes>
              <Route path='/financas/mensal' element={<FinanceMonthly />} />
            </Routes>
          </MemoryRouter>
        </PeriodProvider>
      </AuthProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
