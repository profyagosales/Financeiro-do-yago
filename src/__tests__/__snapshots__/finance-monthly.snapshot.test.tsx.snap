// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FinanceMonthly snapshot > matches structure 1`] = `
<DocumentFragment>
  <div
    class="space-y-6 pb-24"
  >
    <section
      class="rounded-xl p-8 shadow-sm bg-gradient-to-b from-[color:var(--clr)]/80 to-[color:var(--clr)]/10 relative overflow-hidden"
      style="--clr: #6eca00;"
    >
      <h1
        class="text-2xl font-semibold text-white flex items-center gap-3 drop-shadow-sm"
      >
         Finanças — Mensal
      </h1>
      <div
        class="mt-4 space-y-4 text-white/90"
      >
        <div
          class="grid grid-cols-2 lg:grid-cols-7 gap-3 items-end w-full"
        >
          <div
            class="col-span-2"
          >
            <span
              class="mb-1 block text-xs text-white/80"
            >
              Mês
            </span>
            <button
              aria-autocomplete="none"
              aria-controls="radix-«r0»"
              aria-expanded="false"
              aria-label="Mês"
              class="flex w-full items-center justify-between whitespace-nowrap border px-3 py-2 text-sm shadow-sm ring-offset-background focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&>span]:line-clamp-1 data-[placeholder]:text-fg-muted h-10 rounded-xl bg-white/20 text-white border-white/30"
              data-state="closed"
              dir="ltr"
              role="combobox"
              type="button"
            >
              <span
                style="pointer-events: none;"
              >
                agosto/2025
              </span>
              <svg
                aria-hidden="true"
                class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m6 9 6 6 6-6"
                />
              </svg>
            </button>
          </div>
          <div>
            <span
              class="mb-1 block text-xs text-white/80"
            >
              Categoria
            </span>
            <div
              class="flex w-full items-center gap-2 w-full"
            >
              <button
                aria-autocomplete="none"
                aria-controls="radix-«r4»"
                aria-expanded="false"
                aria-label="Categoria"
                class="flex items-center justify-between whitespace-nowrap px-3 py-2 text-sm ring-offset-background focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&>span]:line-clamp-1 text-foreground data-[placeholder]:text-fg-muted w-full h-10 rounded-xl bg-white/70 backdrop-blur border border-white/30 shadow-sm dark:bg-zinc-900/50 dark:border-white/10"
                data-placeholder=""
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  style="pointer-events: none;"
                >
                  Todas
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div>
            <span
              class="mb-1 block text-xs text-white/80"
            >
              Fonte
            </span>
            <div
              class="grid gap-2 w-full"
            >
              <div
                class="inline-flex overflow-hidden rounded-xl border border-white/30 bg-white/70 backdrop-blur shadow-sm"
              >
                <button
                  class="px-3 py-2 text-sm transition-colors bg-emerald-600 text-white"
                  title="Usar conta como fonte"
                  type="button"
                >
                  <span
                    class="inline-flex items-center gap-1"
                  >
                    <svg
                      aria-hidden="true"
                      class="lucide lucide-wallet h-4 w-4"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M19 7V4a1 1 0 0 0-1-1H5a2 2 0 0 0 0 4h15a1 1 0 0 1 1 1v4h-3a2 2 0 0 0 0 4h3a1 1 0 0 0 1-1v-2a1 1 0 0 0-1-1"
                      />
                      <path
                        d="M3 5v14a2 2 0 0 0 2 2h15a1 1 0 0 0 1-1v-4"
                      />
                    </svg>
                     Conta
                  </span>
                </button>
                <button
                  class="px-3 py-2 text-sm transition-colors text-zinc-700"
                  title="Usar cartão como fonte"
                  type="button"
                >
                  <span
                    class="inline-flex items-center gap-1"
                  >
                    <svg
                      aria-hidden="true"
                      class="lucide lucide-credit-card h-4 w-4"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <rect
                        height="14"
                        rx="2"
                        width="20"
                        x="2"
                        y="5"
                      />
                      <line
                        x1="2"
                        x2="22"
                        y1="10"
                        y2="10"
                      />
                    </svg>
                     Cartão
                  </span>
                </button>
              </div>
              <button
                aria-autocomplete="none"
                aria-controls="radix-«rc»"
                aria-expanded="false"
                aria-label="Fonte"
                class="flex items-center justify-between whitespace-nowrap px-3 py-2 text-sm ring-offset-background focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&>span]:line-clamp-1 text-foreground data-[placeholder]:text-fg-muted w-full h-10 rounded-xl bg-white/70 backdrop-blur border border-white/30 shadow-sm"
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  style="pointer-events: none;"
                >
                  Todas
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="col-span-2"
          >
            <span
              class="mb-1 block text-xs text-white/80"
            >
              Buscar
            </span>
            <div
              class="relative"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-search absolute left-3 top-1/2 -translate-y-1/2 h-4 w-4 text-white/60"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m21 21-4.34-4.34"
                />
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
              </svg>
              <input
                aria-label="Buscar"
                class="flex w-full border px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder-fg-muted focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] md:text-sm h-10 pl-9 rounded-xl bg-white/20 text-white placeholder:text-white/50 border-white/30"
                placeholder="Descrição, loja…"
                value=""
              />
            </div>
          </div>
          <div
            class="flex flex-col gap-2"
          >
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 w-full bg-white/10 border-white/30 text-white"
              style="opacity: 0; transform: translateY(-4px);"
              tabindex="0"
            >
              Limpar
            </button>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-8 rounded-md px-3 text-xs"
              style="opacity: 0; transform: translateY(-4px);"
              tabindex="0"
            >
              + Nova transação
            </button>
          </div>
        </div>
      </div>
    </section>
    <section
      class="u-card-base p-3 sm:p-4 flex flex-wrap items-center gap-2 sm:gap-3 justify-between"
    >
      <div
        class="text-xs sm:text-sm text-emerald-900/80"
      >
        0 lançamentos filtrados
      </div>
      <div
        class="flex flex-wrap items-center gap-2 sm:gap-3"
      >
        <button
          class="justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2 inline-flex items-center gap-2"
          style="opacity: 0; transform: translateY(-4px);"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-plus h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12h14"
            />
            <path
              d="M12 5v14"
            />
          </svg>
           Nova transação
        </button>
        <button
          class="justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 inline-flex items-center gap-2"
          style="opacity: 0; transform: translateY(-4px);"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-download h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 15V3"
            />
            <path
              d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
            />
            <path
              d="m7 10 5 5 5-5"
            />
          </svg>
           Exportar PDF
        </button>
        <button
          class="justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-9 px-4 py-2 inline-flex items-center gap-2"
          style="opacity: 0; transform: translateY(-4px);"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-copy h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="14"
              rx="2"
              ry="2"
              width="14"
              x="8"
              y="8"
            />
            <path
              d="M4 16c-1.1 0-2-.9-2-2V4c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2"
            />
          </svg>
           Duplicar
        </button>
      </div>
    </section>
    <div
      class="grid gap-4 sm:grid-cols-2 xl:grid-cols-4"
    >
      <div
        aria-live="polite"
        class="animate-pulse rounded-md bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-24 w-full"
        role="status"
      />
      <div
        aria-live="polite"
        class="animate-pulse rounded-md bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-24 w-full"
        role="status"
      />
      <div
        aria-live="polite"
        class="animate-pulse rounded-md bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-24 w-full"
        role="status"
      />
      <div
        aria-live="polite"
        class="animate-pulse rounded-md bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-24 w-full"
        role="status"
      />
    </div>
    <section
      class="grid gap-6 lg:grid-cols-12"
    >
      <div
        class="lg:col-span-6"
      >
        <div
          aria-live="polite"
          class="animate-pulse rounded-md bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-72 w-full"
          role="status"
        />
      </div>
      <div
        class="lg:col-span-6"
      >
        <div
          aria-live="polite"
          class="animate-pulse rounded-md bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-72 w-full"
          role="status"
        />
      </div>
    </section>
    <div
      aria-live="polite"
      class="animate-pulse rounded-md bg-[linear-gradient(90deg,#ececec,#f5f5f5,#ececec)] bg-[length:800px_100%] h-64 w-full"
      role="status"
    />
    <button
      aria-label="Nova transação"
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:bg-[hsl(var(--disabled-bg))] disabled:text-[hsl(var(--disabled-fg))] [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground hover:bg-primary/90 px-4 py-2 fixed bottom-6 right-6 md:bottom-8 md:right-8 z-40 w-14 h-14 rounded-full shadow-xl"
      data-state="closed"
      style="opacity: 0; transform: translateY(-4px);"
      tabindex="0"
      title="Nova transação"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-plus h-6 w-6"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 12h14"
        />
        <path
          d="M12 5v14"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
